packages/fether-react/src/Send/SignedTxSummary/SignedTxSummary.js

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import React, { Component } from 'react';
+import { Field, Form } from 'react-final-form';
+import { Form as FetherForm, Header } from 'fether-ui';
+import { inject, observer } from 'mobx-react';
+import { Link } from 'react-router-dom';
+import { withProps } from 'recompose';
+
+import RequireHealth from '../../RequireHealthOverlay';
+import TokenBalance from '../../Tokens/TokensList/TokenBalance';
+import withAccount from '../../utils/withAccount';
+import withBalance, { withEthBalance } from '../../utils/withBalance';
+import withTokens from '../../utils/withTokens';
+
+@inject('parityStore', 'sendStore')
+@withTokens
+@withProps(({ match: { params: { tokenAddress } }, tokens }) => ({
+  token: tokens[tokenAddress]
+}))
+@withAccount
+@withBalance // Balance of current token (can be ETH)
+@withEthBalance // ETH balance
+@observer
+class SignedTxSummary extends Component {
+  state = {
+    maxSelected: false
+  };
+  handleSubmit = values => {
+    const {
+      account: { address },
+      history,
+      sendStore,
+      token
+    } = this.props;
+
+    return sendStore
+      .sendRaw()
+      .then(() => history.push(`/send/${token.address}/from/${address}/sent`))
+      .catch(error => ({
+        password: error.text
+      }));
+  };
+
+  render () {
+    const {
+      account: { address },
+      sendStore: { tx },
+      token
+    } = this.props;
+
+    return (
+      <div>
+        <Header
+          left={
+            <Link to={`/tokens/${address}`} className='icon -back'>
+              Close
+            </Link>
+          }
+          title={token && <h1>Send {token.name}</h1>}
+        />
+
+        <RequireHealth require='sync'>
+          <div className='window_content'>
+            <div className='box -padded'>
+              <TokenBalance
+                decimals={6}
+                drawers={[
+                  <Form
```

Comment:
It would make sense to factorize code between this and TxForm, if it's easily doable